packages/ckeditor5-table/tests/tablecellproperties/tablecellpropertiesediting.js

```diff
@@ -1241,6 +1242,111 @@ describe( 'table cell properties', () => {
+
+				editor = await VirtualTestEditor.create( {
+					plugins: [ TableCellPropertiesEditing, Paragraph, TableEditing, TableLayoutEditing ]
+				} );
+
+				model = editor.model;
+			} );
+
+			describe( 'upcast', () => {
+				describe( 'horizontal alignment', () => {
+					it( 'should not upcast text-align:left style (due to the default value of the property)', () => {
+						editor.setData( '<table class="layout-table"><tr><td style="text-align:left">foo</td></tr></table>' );
+						const tableCell = model.document.getRoot().getNodeByPath( [ 0, 0, 0 ] );
+
+						expect( tableCell.getAttribute( 'tableCellHorizontalAlignment' ) ).to.be.undefined;
+					} );
+
+					it( 'should upcast text-align:right style', () => {
+						editor.setData( '<table class="layout-table"><tr><td style="text-align:right">foo</td></tr></table>' );
+						const tableCell = model.document.getRoot().getNodeByPath( [ 0, 0, 0 ] );
+
+						expect( tableCell.getAttribute( 'tableCellHorizontalAlignment' ) ).to.equal( 'right' );
+					} );
+
+					describe( 'the [align] attribute', () => {
+						it( 'should not upcast the align=left attribute (due to the default value of the property)', () => {
+							editor.setData( '<table class="layout-table"><tr><td align="left">foo</td></tr></table>' );
+							const tableCell = model.document.getRoot().getNodeByPath( [ 0, 0, 0 ] );
+
+							expect( tableCell.getAttribute( 'tableCellHorizontalAlignment' ) ).to.be.undefined;
+						} );
+
+						it( 'should upcast the align=right attribute ', () => {
+							editor.setData( '<table class="layout-table"><tr><td align="right">foo</td></tr></table>' );
+							const tableCell = model.document.getRoot().getNodeByPath( [ 0, 0, 0 ] );
+
+							expect( tableCell.getAttribute( 'tableCellHorizontalAlignment' ) ).to.be.equal( 'right' );
+						} );
+
+						it( 'should upcast the align=center attribute', () => {
+							editor.setData( '<table class="layout-table"><tr><td align="center">foo</td></tr></table>' );
+							const tableCell = model.document.getRoot().getNodeByPath( [ 0, 0, 0 ] );
+
+							expect( tableCell.getAttribute( 'tableCellHorizontalAlignment' ) ).to.equal( 'center' );
+						} );
+
+						it( 'should upcast the align=justify attribute', () => {
+							editor.setData( '<table class="layout-table"><tr><td align="justify">foo</td></tr></table>' );
+							const tableCell = model.document.getRoot().getNodeByPath( [ 0, 0, 0 ] );
+
+							expect( tableCell.getAttribute( 'tableCellHorizontalAlignment' ) ).to.equal( 'justify' );
+						} );
+					} );
+				} );
+
+				describe( 'upcast conversion', () => {
+					it( 'should upcast "top" vertical-align', () => {
+						editor.setData( '<table class="layout-table><tr><td style="vertical-align:top">foo</td></tr></table>' );
+
+						const tableCell = model.document.getRoot().getNodeByPath( [ 0, 0, 0 ] );
+
+						expect( tableCell.getAttribute( 'tableCellVerticalAlignment' ) ).to.equal( 'top' );
+					} );
```

Comment:
This test fails. Pretty unexpected behavior. It outputs such HTML:

```html
<p>foo</p><figure class="table layout-table" role="presentation"><table><tbody><tr><td>&nbsp;</td></tr></tbody></table></figure>
```

So `foo` is being moved to the paragraph outside the table. It works fine for content tables.